test/unicode/types.jl

```diff
@@ -0,0 +1,11 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+nullstring16 = UInt16[]
+badstring16  = Uint16[0x0065]
+@test_throws UnicodeError UTF16String(nullstring16)
+@test_throws UnicodeError UTF16String(badstring16)
+
+nullstring32 = UInt32[]
+badstring32  = Uint32['a']
```

Comment:
Not a huge deal since there are still (silent) deprecations for them I think, can fix it along with the next coverage PR.